phasespace/fulldecay/fulldecay.py

```diff
@@ -0,0 +1,259 @@
+import itertools
+from typing import Callable, Union
+
+import tensorflow as tf
+import tensorflow.experimental.numpy as tnp
+from particle import Particle
+
+from phasespace import GenParticle
+
+from .mass_functions import _DEFAULT_CONVERTER
+
+_MASS_WIDTH_TOLERANCE = 0.01
+_DEFAULT_MASS_FUNC = "rel-BW"
+
+
+class FullDecay:
+    """A container that works like GenParticle that can handle multiple decays."""
+
+    def __init__(self, gen_particles: list[tuple[float, GenParticle]]):
+        """Create an instance of FullDecay.
+
+        Parameters
+        ----------
+        gen_particles : list[tuple[float, GenParticle]]
+            All the GenParticles and their corresponding probabilities.
+            The list must be of the format [[probability, GenParticle instance], [probability, ...
+        Notes
+        -----
+        Input format might change
+        """
+        self.gen_particles = gen_particles
+
+    @classmethod
+    def from_dict(
+        cls,
+        dec_dict: dict,
+        mass_converter: dict[str, Callable] = None,
+        tolerance: float = _MASS_WIDTH_TOLERANCE,
+    ):
+        """Create a FullDecay instance from a dict in the decaylanguage format.
+
+        Parameters
+        ----------
+        dec_dict : dict
+            The input dict from which the FullDecay object will be created from.
+        mass_converter : dict[str, Callable]

```

Comment:
This is another place where I would advocate that the user will hugely benefit from an example of such a mass converter dict.

Reply:
BTW default values should be in the description, see again for example https://github.com/scikit-hep/particle/blob/master/src/particle/particle/particle.py#L433-L537. (It is a detail that you use another style of documentation - Google vs NumPy styles.)